amr-wind/equation_systems/tke/source_terms/KransAxell.cpp

```diff
@@ -0,0 +1,126 @@
+#include <AMReX_Orientation.H>
+
+#include "amr-wind/equation_systems/tke/source_terms/KransAxell.H"
+#include "amr-wind/CFDSim.H"
+#include "amr-wind/turbulence/TurbulenceModel.H"
+
+namespace amr_wind::pde::tke {
+
+KransAxell::KransAxell(const CFDSim& sim)
+    : m_turb_lscale(sim.repo().get_field("turb_lscale"))
+    , m_shear_prod(sim.repo().get_field("shear_prod"))
+    , m_buoy_prod(sim.repo().get_field("buoy_prod"))
+    , m_dissip(sim.repo().get_field("dissipation"))
+    , m_tke(sim.repo().get_field("tke"))
+    , m_time(sim.time())
+    , m_sim(sim)
+    , m_mesh(sim.mesh())
+    , m_velocity(sim.repo().get_field("velocity"))
+{
+    AMREX_ALWAYS_ASSERT(sim.turbulence_model().model_name() == "KLAxell");
+    auto coeffs = sim.turbulence_model().model_coeffs();
+    amrex::ParmParse pp("ABL");
+    pp.query("reference_temperature", m_ref_temp);
+    pp.query("surface_temp_flux", m_heat_flux);
+    pp.query("meso_sponge_start", m_sponge_start);
+    {
+        amrex::ParmParse pp_incflow("incflo");
+        pp_incflow.queryarr("gravity", m_gravity);
+    }
+}
+
+KransAxell::~KransAxell() = default;
+
+void KransAxell::operator()(
+    const int lev,
+    const amrex::MFIter& mfi,
+    const amrex::Box& bx,
+    const FieldState fstate,
+    const amrex::Array4<amrex::Real>& src_term) const
+{
+    const auto& vel =
+        m_velocity.state(field_impl::dof_state(fstate))(lev).const_array(mfi);
+    const auto& tlscale_arr = (this->m_turb_lscale)(lev).array(mfi);
+    const auto& shear_prod_arr = (this->m_shear_prod)(lev).array(mfi);
+    const auto& buoy_prod_arr = (this->m_buoy_prod)(lev).array(mfi);
+    const auto& dissip_arr = (this->m_dissip)(lev).array(mfi);
+    const auto& tke_arr = m_tke(lev).array(mfi);
+    const auto& geom = m_mesh.Geom(lev);
+    const auto& problo = m_mesh.Geom(lev).ProbLoArray();
+    const auto& probhi = m_mesh.Geom(lev).ProbHiArray();
+    const auto& dx = geom.CellSizeArray();
+    const auto& dt = m_time.delta_t();
+    const amrex::Real ref_temp = m_ref_temp;
+    const amrex::Real heat_flux =
+        std::abs(m_gravity[2]) / ref_temp * m_heat_flux;
+    const amrex::Real Cmu = 0.556;
+    const amrex::Real sponge_start = m_sponge_start;
+    const amrex::Real ref_tke = m_ref_tke;
+    const auto tiny = std::numeric_limits<amrex::Real>::epsilon();
+    const amrex::Real kappa = 0.41;
+    const amrex::Real z0 = 0.1;
```

Comment:
Is this z0 supposed to be hard coded for the model?

Reply:
Does not have to be. I will modify to read it from ABL.